### ISSUE TITLE
CS9169 is not reported for record structs with InlineArray attribute

### ISSUE THREAD

**Issue by tmat** [dotnet/roslyn#69402]
```C#
[InlineArray(2)]
public record struct Buffer(int a, int b);  // no error

[InlineArray(2)]
public struct Buffer(int b, int b); //  error CS9169: Inline array struct must declare one and only one instance field.
```

**Comment by tmat** [comment]
Odd the error is reported by the build but not in the IDE :-o

**Comment by jcouv** [comment]
> Is it an emit error that's not reported to Error List?

I don't think so. `IsBuildOnlyDiagnostic(ERR_InvalidInlineArrayFields)` returns false.
Can you check what compiler versions are being used (add `#error version` in your source) to ensure that the same compiler bits are used in both scenarios?

**Comment by jcouv** [comment]
FWIW, I'm observing the same IDE vs. Build differences... I verified that both use the same compiler version.
Let's separate issues. 

@tmat, as far as I can tell, the Build behavior is correct. Is that also your understanding?

**Comment by CyrusNajmabadi** [comment]
Closing out.  I don't see any issues here with the diagnostics reported.
